automation_tools/satellite6/capsule.py

```diff
@@ -0,0 +1,398 @@
+        """Return a host_string in the format expected by Fabric"""
+        return '{}@{}:{}'.format(self.user, self.hostname, self.port)
+
+
+class Config(object):
+    """Configuration information provide easy access to configuration and some
+    helper methods to identify is some configuration is present or not.
+
+    """
+    def __init__(self, path):
+        self.path = path
+        self.organization_label = None
+        self.environment = None
+        self.content_view = None
+        self.activation_key = None
+        self.admin_user = None
+        self.admin_password = None
+        self.defaults = None
+        self.server = None
+        self.capsules = []
+        self._key_filenames = set()
+
+        self._parse()
+
+    def _parse(self):
+        """Parse the configuration and store the contents"""
+        with open(self.path) as handler:
+            data = json.load(handler)
+        self.organization_label = data.get('organization-label')
+        self.environment = data.get('environment')
+        self.content_view = data.get('content-view')
+        self.activation_key = data.get('activation-key')
+        self.admin_user = data.get('admin-user')
+        self.admin_password = data.get('admin-password')
+        defaults = data.get('defaults')
+        if defaults is not None and isinstance(defaults, dict):
+            key_filename = defaults.get('key-filename')
+            self._key_filenames.add(key_filename)
+            self.defaults = Credentials(
+                user=defaults.get('user'),
+                password=defaults.get('password'),
+                key_filename=key_filename,
+            )
+        server = data.get('server')
+        if server is not None and isinstance(server, dict):
+            key_filename = server.get('key-filename')
+            self._key_filenames.add(key_filename)
+            self.server = HostConfig(
+                hostname=server.get('hostname'),
+                user=server.get('user', self.defaults.user),
+                password=server.get('password', self.defaults.password),
+                key_filename=key_filename,
+            )
+        capsules = data.get('capsules')
+        if capsules is not None and isinstance(capsules, list):
+            for capsule in capsules:
+                if capsule is not None and isinstance(capsule, dict):
+                    key_filename = capsule.get('key-filename')
+                    self._key_filenames.add(key_filename)
+                    self.capsules.append(HostConfig(
+                        hostname=capsule.get('hostname'),
+                        user=capsule.get('user', self.defaults.user),
+                        password=capsule.get(
+                            'password', self.defaults.password),
+                        key_filename=key_filename,
+                    ))
+
+    @property
+    def key_filenames(self):
+        """Return a list of collect key filenames or None if the list is
+        empty.
```

Comment:
None is the default value for that config on Fabric, so this method is ensuring that.